src/queryweaver/async_client.py

```
@@ -0,0 +1,385 @@
+            ValueError: If database not loaded or query is empty
+            RuntimeError: If SQL generation fails
+        """
+        # Use base class validation
+        self._validate_query_params(database_name, query)
+        
+        # Use base class helper to prepare chat data
+        chat_data = self._prepare_chat_data(query, instructions, chat_history)
+        
+        try:
+            result = await self._generate_sql_async(database_name, chat_data)
+            return result
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)